sonarts-core/src/utils/nodes.ts

```diff
@@ -0,0 +1,196 @@
+  return is(node, ts.SyntaxKind.ExpressionStatement);
+}
+
+export function isIdentifier(node: ts.Node): node is ts.Identifier {
+  return is(node, ts.SyntaxKind.Identifier);
+}
+
+export function isInterfaceDeclaration(node: ts.Node): node is ts.InterfaceDeclaration {
+  return is(node, ts.SyntaxKind.InterfaceDeclaration);
+}
+
+export function isIfStatement(node: ts.Node): node is ts.IfStatement {
+  return is(node, ts.SyntaxKind.IfStatement);
+}
+
+export function isLabeledStatement(node: ts.Node): node is ts.LabeledStatement {
+  return is(node, ts.SyntaxKind.LabeledStatement);
+}
+
+export function isLiteralExpression(node: ts.Node): node is ts.LiteralExpression {
+  return ts.SyntaxKind.FirstLiteralToken <= node.kind && node.kind <= ts.SyntaxKind.LastLiteralToken;
+}
+
+export function isNumericLiteral(node: ts.Node): node is ts.NumericLiteral {
+  return is(node, ts.SyntaxKind.NumericLiteral);
+}
+
+export function isObjectBindingPattern(node: ts.Node): node is ts.ObjectBindingPattern {
+  return is(node, ts.SyntaxKind.ObjectBindingPattern);
+}
+
+export function isObjectLiteralExpression(node: ts.Node): node is ts.ObjectLiteralExpression {
+  return is(node, ts.SyntaxKind.ObjectLiteralExpression);
+}
+
+export function isParenthesizedExpression(node: ts.Node): node is ts.ParenthesizedExpression {
+  return is(node, ts.SyntaxKind.ParenthesizedExpression);
+}
+
+export function isPostfixUnaryExpression(node: ts.Node): node is ts.PostfixUnaryExpression {
+  return is(node, ts.SyntaxKind.PostfixUnaryExpression);
+}
+
+export function isPrefixUnaryExpression(node: ts.Node): node is ts.PrefixUnaryExpression {
+  return is(node, ts.SyntaxKind.PrefixUnaryExpression);
+}
+
+export function isPropertyAccessExpression(node: ts.Node): node is ts.PropertyAccessExpression {
+  return is(node, ts.SyntaxKind.PropertyAccessExpression);
+}
+
+export function isPropertyAssignment(node: ts.Node): node is ts.PropertyAssignment {
+  return is(node, ts.SyntaxKind.PropertyAssignment);
+}
+
+export function isReturnStatement(node: ts.Node): node is ts.ReturnStatement {
+  return is(node, ts.SyntaxKind.ReturnStatement);
+}
+
+export function isShorthandPropertyAssignment(node: ts.Node): node is ts.ShorthandPropertyAssignment {
+  return is(node, ts.SyntaxKind.ShorthandPropertyAssignment);
+}
+
+export function isSpreadAssignment(node: ts.Node): node is ts.SpreadAssignment {
+  return is(node, ts.SyntaxKind.SpreadAssignment);
+}
+
+export function isSpreadElement(node: ts.Node): node is ts.SpreadElement {
+  return is(node, ts.SyntaxKind.SpreadElement);
+}
+
+export function isStringLiteral(node: ts.Node): node is ts.StringLiteral {
+  return is(node, ts.SyntaxKind.StringLiteral);
+}
+
+export function isThrowStatement(node: ts.Node): node is ts.ThrowStatement {
+  return is(node, ts.SyntaxKind.ThrowStatement);
+}
+
+export function isTypeNode(node: ts.Node): node is ts.TypeNode {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: The function isTypeNode has a cyclomatic complexity of 21 which is higher than the threshold of 10 ([typescript:S1541](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1541))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=stas%2Ftype-guards&issues=AWC9a1es_Q-0dz4JA1kF&open=AWC9a1es_Q-0dz4JA1kF)

[ISSUE_KEY]: AWC9a1es_Q-0dz4JA1kF

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED